MetabolicAtlas/MetabolicAtlas
# feat: add support for fluxomics in data overlay

**Related issue(s) and PR(s)**  
This PR closes #1066  and relates to private issue 138. Joint work with @e0 .


Adds functionality for data overlay for fluxomics.

**Type of change**  


 
**List of changes made**  

- added data type component in mapping in `frontend/src/helpers/dataOverlay.js`


**Screenshot of the fix**  

![2022-08-15-141959_941x635_scrot](https://user-images.githubusercontent.com/1029190/184634034-acdbd10e-35ef-4cba-85b5-0d49b930bc26.png)


**Testing**  

- A certain branch in other repos is needed
- A rebuild is needed due to new dependencies
- Relative urls that can be reused both for production and local testing
- Instructions on how to test
1. Get the latest changes from [`data-files`](https://github.com/MetabolicAtlas/data-files/pull/31) and [`data-generation`](https://github.com/MetabolicAtlas/data-generation/pull/32).
2. Rebuild and restart the stack `build-stack && start-stack`.
3. Visit `/explore/Human-GEM/map-viewer/golgi_apparatus`. Expand the Data Overlay section. Verify that `fluxomics` is available under `Select data type`.
4. Select a level under `Levels from Human Protein Atlas 21.0`. Verify that the reactions in the map get re-colored.
5. Please also test on the `3D` map.
6. Upload custom expression data (example below) including reactions. Verify that the reactions in the map get re-colored.
7. Visit `http://localhost/documentation#data-overlay`. Verify that the documentation is updated and is correct.

`reaction1.mock.tsv`
```tsv
id	tissue 1	tissue 2
MAR01591	0.6	0.3
MAR00447	0.55	0.9
```

**Further comments**  


**Definition of Done checklist**
